app/domain/authentication/resource_restrictions/resource_restrictions.rb

```diff
@@ -0,0 +1,25 @@
+module Authentication
+  module ResourceRestrictions
+
+    class ResourceRestrictions
+
+      def initialize(resource_restrictions_hash:)
+        @resource_restrictions_hash = resource_restrictions_hash
+      end
+
+      def names
+        @resource_restrictions_hash.keys
+      end
+
+      def each(&block)
```

Comment:
what is &block here?

Reply:
This is a propagator for `each` method on enumerable.
This is how you define a method that accepts a block (everything that comes in a curly braces after method call):
```
[1, 2, 3].each {|n| puts n}
```

Reply:
oh didn't know that!

Why would you use this function over just using `hash.each({...})` in-place for a all ResourceRestrictions instances?